### PR TITLE
Generate include/capstone/support.h

### PR DESCRIPTION
**Pull request by radare**
Fix https://github.com/aquynh/capstone/issues/508

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -359,6 +360,15 @@ ifeq ($(CAPSTONE_SHARED),yes)
...
 endif
 
+$(SUPPORT):
+	echo '#ifndef CAPSTONE_SUPPORT_H' > $@
+	echo '#define CAPSTONE_SUPPORT_H' >> $@
+	for a in $(CAPSTONE_ARCHS) ; do \
+		ARCH=$$(echo $$a|tr a-z A-Z); \
+		echo '#define CS_SUPPORT_ARCH_'$${ARCH} 1 >> $@ ; \
+	done
+	echo '#endif' >> $@
```
Comment (nplanel) [inline Makefile]: You should add the API version too, encoded as a flat integer like #define CAPSTONE_VERSION(a,b,c) ((a)<<24+(b)<<16+(c)<<8)

Comment (nplanel) [inline Makefile]: What about cmake files ?

Comment (radare) [inline Makefile]: api version is already available in capstone.h via the MAJOR/MINOR defines

Comment (radare) [inline Makefile]: i dont get the point of using cmake here. make is quite legacy and for a simple project like this we dont need cmake or autotools

**Comment by aquynh** [comment]
make is not well supported outside of *nix world. example is Windows with MSVC. CMake is helpful in such a case.

**Comment by radare** [comment]
In addition to this check the code that uses a specific arch for capstone should begin with:
```
#ifdef CAPSTONE_M68K_H
...
```
this means that the compile-time version of capstone have this include, but this doesn't means that the archive file contains them. So this must be checked using the CS_SUPPORT vars. The problem I see is with the backward compatibility of my approach.

Maybe it will be better to just generate a `capstone.h` with only the #includes of the archs that have been selected at compile time?

**Comment by radare** [comment]
this got stuck in time and needs to be rebased, can you reply my questions before spending time in the rebase?

**Comment by radare** [comment]
hello? anyone is there?

**Comment by XVilka** [comment]
@radare maybe close this?

**Pull request closed without merging**